docs/tutorial/02-draw-a-snake.md

```diff
@@ -99,47 +99,61 @@ defmodule Snake.Scene.Game do
+Running the game again, we see we have some information about the size of the viewport in pixels. We can use this information to compute the number of rows and columns in our grid. We'll define the default length of 32 pixels for the edge of one square and then compute how many we can fit onto the screen. As this information is necessary throughout the whole game, we store it in the `state` of the scene.
 
-When running the game you may have noticed the compiler warning us:
+```elixir
+@tile_size 32
 
-    warning: variable "arg" is unused (if the variable is not meant to be used, prefix it with an underscore)
+def init(arg, opts) do
+  viewport = opts[:viewport]
 
-In our configuration, we are passing `nil` as the first `arg` parameter to the `init/2` function of our `Game` scene, which already indicates that we're not planning to provide any information meant for further usage here. So let's follow the compiler's suggestions and prefix the argument with an underscore to indicate that we know about the parameter but don't need it:
+  {:ok, %ViewPort.Status{size: {vp_width, vp_height}}} = ViewPort.info(viewport)
 
-```elixir
-def init(_arg, opts) do
-  viewport = opts[:viewport]
+  number_of_columns = div(vp_width, @tile_size)
+  number_of_rows = div(vp_height, @tile_size)
 
-  IO.inspect(ViewPort.info(viewport), label: "viewport info")
+  state = %{
+    width: number_of_columns,
+    height: number_of_rows
+  }
+
+  {:ok, :ok}
 end
 ```
 
-Now let's use the view port information to set up our grid. We'll define the default length of 32 pixels for the edge of one square and then compute how many we can fit onto the screen. As this information is necessary throughout the whole game, we store it in the `state` of the scene.
+> Coach: Talk about assignments that use pattern matching
 
-```elixir
-@tile_size 32
+When running the game you may have noticed the compiler warning us:
 
+    warning: variable "arg" is unused (if the variable is not meant to be used, prefix it with an underscore)
+
+In our configuration, we are passing `nil` as the argument of the first parameter `arg` of the `init/2` function of our `Game` scene, which already indicates that we're not planning to provide any information meant for further usage here. So let's follow the compiler's suggestions and prefix the parameter with an underscore to indicate that we know about the parameter but don't need it:
+
+```elixir
 def init(_arg, opts) do
   viewport = opts[:viewport]
 
   {:ok, %ViewPort.Status{size: {vp_width, vp_height}}} = ViewPort.info(viewport)
 
-  num_tiles_width = trunc(vp_width / @tile_size)
-  num_tiles_height = trunc(vp_height / @tile_size)
+  number_of_columns = div(vp_width, @tile_size)
+  number_of_rows = div(vp_height, @tile_size)
 
   state = %{
-    width: num_tiles_width,
-    height: num_tiles_height
+    width: number_of_columns,
+    height: number_of_rows
   }
+
+  {:ok, :ok}
```

Comment:
Hm, just wondering: doesn't the compiler also complain if we don't use the `state` any further here?

On the other hand, it's sort of an intermediate step where we don't even encourage people to run it. So probably alright like this.

Reply:
> Hm, just wondering: doesn't the compiler also complain if we don't use the state any further here?

yes, it does

> On the other hand, it's sort of an intermediate step where we don't even encourage people to run it. So probably alright like this.

That was my thought too